Lib/test/test_readline.py

```diff
@@ -127,6 +127,13 @@ def test_init(self):
 input()
 print("History length:", readline.get_current_history_length())
 """
+    auto_history_script_with_null = """\
+import readline
+readline.set_auto_history({})
```

Comment:
I suspect the _readline_ module is not directly related at all. Perhaps the test would be better off in _test_builtin.py_. There are already some tests there in the _PtyTests_ class that call _input_ on a terminal, so you may be able to use them for inspiration.